crates/driver/src/infra/observe/mod.rs

```diff
@@ -348,6 +348,35 @@ pub fn mempool_executed(
         .mempool_submission
         .with_label_values(&[&mempool.to_string(), result])
         .inc();
+
+    // For some of the errors we are interested in observing the exact block numbers
+    // passed since the first submission.
+    let blocks_passed = match res {
+        Ok(_) => None,
```

Comment:
Even in the ok case we should measure how many blocks have passed before the transaction was included.